app_test_suite/steps/gotest/gotest.py

```diff
@@ -0,0 +1,181 @@
+
+from app_test_suite.cluster_manager import ClusterManager
+from app_test_suite.cluster_providers.cluster_provider import ClusterInfo
+from app_test_suite.errors import ATSTestError
+from app_test_suite.steps.base_test_runner import BaseTestScenario
+from app_test_suite.steps.base_test_runner import (
+    TestInfoProvider,
+    context_key_chart_yaml,
+    TestExecInfo,
+    TestExecutor,
+    BaseTestScenariosFilteringPipeline,
+)
+from app_test_suite.steps.test_types import STEP_TEST_SMOKE, STEP_TEST_FUNCTIONAL
+from app_test_suite.steps.upgrade_test_runner import BaseUpgradeTestScenario
+
+logger = logging.getLogger(__name__)
+
+
+class GotestTestFilteringPipeline(BaseTestScenariosFilteringPipeline):
+    key_config_option_gotest_dir = "--app-tests-gotest-tests-dir"
+
+    def __init__(self) -> None:
+        cluster_manager = ClusterManager()
+        super().__init__(
+            [
+                TestInfoProvider(),
+                GotestSmokeTestScenario(cluster_manager),
+                GotestFunctionalTestScenario(cluster_manager),
+                GotestUpgradeTestScenario(cluster_manager),
+            ],
+            cluster_manager,
+        )
+
+    def initialize_config(self, config_parser: configargparse.ArgParser) -> None:
+        super().initialize_config(config_parser)
+        self._config_parser_group = cast(
+            configargparse.ArgParser,
+            config_parser.add_argument_group("Gotest specific options"),
+        )
+        self._config_parser_group.add_argument(
+            self.key_config_option_gotest_dir,
+            required=False,
+            default=os.path.join("tests", "ats"),
+            help="Directory, where go tests source code can be found.",
+        )
+
+
+class GotestExecutorMixin(TestExecutor):
+    _GOTEST_BIN = "go"
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        # This class is intended to be used as a mixin, that forwards constructor call to any other type
+        #  inherited from except the mixin itself.
+        super().__init__(*args, **kwargs)  # type: ignore
+        self._gotest_dir = ""
+
+    def prepare_test_environment(self, exec_info: TestExecInfo) -> None:
+        return
+
+    def execute_test(self, exec_info: TestExecInfo) -> None:
+        env_vars = os.environ.copy()
+        env_vars["ATS_CHART_PATH"] = exec_info.chart_path
+        env_vars["ATS_CHART_VERSION"] = exec_info.chart_ver
+        env_vars["ATS_CLUSTER_TYPE"] = exec_info.cluster_type
+        env_vars["ATS_CLUSTER_VERSION"] = exec_info.cluster_version
+        env_vars["ATS_KUBE_CONFIG_PATH"] = exec_info.kube_config_path
+        env_vars["ATS_TEST_TYPE"] = exec_info.test_type
+        env_vars["ATS_TEST_DIR"] = exec_info.test_dir
+
+        if exec_info.app_config_file_path is not None:
+            env_vars["ATS_APP_CONFIG_FILE_PATH"] = exec_info.app_config_file_path
+
+        args = [
+            self._GOTEST_BIN,
+            "test",
+            "-v",
```

Comment:
Can we make this configurable? Running with `v` always can be quite noisy.

Reply:
Yes sure. I added the `-v` because have this in the integration-test job. Good to have it configurable. 👍

Reply:
@MarcelMue Without the `-v` you don't get any output till the test finishes. For more complex tests I think it's useful to see what's going on.